src/main/java/com/google/sps/servlets/ShadowLoadInterviewsServlet.java

```diff
@@ -0,0 +1,210 @@
+import com.google.sps.data.ScheduledInterview;
+import com.google.sps.data.ScheduledInterviewDao;
+import com.google.sps.data.TimeRange;
+import java.io.IOException;
+import java.io.BufferedReader;
+import java.lang.Integer;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.TextStyle;
+import java.util.EnumSet;
+import java.util.Locale;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.HashSet;
+import java.util.Set;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.ServletContext;
+import javax.servlet.ServletException;
+import javax.servlet.RequestDispatcher;
+import java.util.Optional;
+
+@WebServlet("/shadow-load-interviews")
+public class ShadowLoadInterviewsServlet extends HttpServlet {
+  private ScheduledInterviewDao scheduledInterviewDao;
+  private PersonDao personDao;
+  private Instant currentTime;
+  private final int maxTimezoneOffsetMinutes = 720;
+  private final int maxTimezoneOffsetHours = 12;
+
+  @Override
+  public void init() {
+    init(new DatastoreScheduledInterviewDao(), new DatastorePersonDao(), Instant.now());
+  }
+
+  public void init(
+      ScheduledInterviewDao scheduledInterviewDao, PersonDao personDao, Instant currentTime) {
+    this.scheduledInterviewDao = scheduledInterviewDao;
+    this.personDao = personDao;
+    this.currentTime = currentTime;
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    int timezoneOffsetMinutes = Integer.parseInt(request.getParameter("timeZoneOffset"));
+    Preconditions.checkArgument(
+        Math.abs(timezoneOffsetMinutes) <= maxTimezoneOffsetMinutes,
+        "Offset greater than %d minutes (%d hours): %d",
+        maxTimezoneOffsetMinutes,
+        maxTimezoneOffsetHours,
+        timezoneOffsetMinutes);
+    ZoneOffset timezoneOffset = convertIntToOffset(timezoneOffsetMinutes);
+    ZonedDateTime day = generateDay(currentTime, timezoneOffset);
+    ZonedDateTime utcTime = day.withZoneSameInstant(ZoneOffset.UTC);
+    // The user will be shown available interview times for the next four weeks, starting from the
+    // current time.
+    TimeRange interviewSearchTimeRange =
+        new TimeRange(utcTime.toInstant(), utcTime.toInstant().plus(27, ChronoUnit.DAYS));
+    String position = request.getParameter("position");
+    Job selectedPosition = Job.valueOf(Job.class, position);
+    UserService userService = UserServiceFactory.getUserService();
+    String userEmail = userService.getCurrentUser().getEmail();
+    String userId = userService.getCurrentUser().getUserId();
+    // Since Users returned from the LocalUserService (in tests) do not have userIds, here we set
```

Comment:
This hack, particularly the repeated copy pasting of it, is probably the worst thing we've allowed in this code.

(Just a long term lesson -- this project is _full_ of terrible stuff that we've permitted for the sake of expediency. Please don't reflect on this experience as setting the bar for code quality!!).

Reply:
Yeah, we probably should have come up with a better way to set current user in tests earlier on. Basically making our own mock of user service.